src/App/BooksRoute/UpdateBookForm/UpdateBookForm.tsx

```diff
@@ -1,28 +1,46 @@
 import { useState, type FC } from "react";
+import { useMutation, useQuery } from "@apollo/client/react";
 import {
   UpdateBookFormBookDoc,
   UpdateBookFormUpdateBookDoc,
 } from "./UpdateBookForm.graphql";
-import { useMutation, useQuery } from "@apollo/client";
 
 export const UpdateBookForm: FC<{ bookId: string; onClose: () => void }> = ({
   bookId,
   onClose,
 }) => {
-  const [formState, setFormState] = useState<{ isbn: string }>({
-    isbn: "",
-  });
-  const queryBookResult = useQuery(UpdateBookFormBookDoc, {
-    variables: {
-      id: bookId,
-    },
+  const { data, error, loading } = useQuery(UpdateBookFormBookDoc, {
+    variables: { id: bookId },
     fetchPolicy: "cache-and-network",
-    onCompleted: (data) => {
-      if (data.book.__typename === "BookResultOk" && data.book.result) {
-        setFormState({ isbn: data.book.result.isbn });
-      }
-    },
```

Comment:
`onCompleted` is removed in v4: https://www.apollographql.com/docs/react/migrating/apollo-client-4-migration#removal-of-the-oncompleted-and-onerror-callbacks

And surprisingly, the code makes more sense/cleaner now:
1. After the query finishes...
2. Load the result into state using `useState` in a subsequent component, instead of `useEffect`
3. All update functionalities are in the subsequent component